static/app/views/performance/newTraceDetails/traceApi/utils.tsx

```diff
@@ -8,3 +16,81 @@ export function isEmptyTrace(trace: TraceTree.Trace): boolean {
 
   return trace.length === 0;
 }
+
+const CANDIDATE_TRACE_TITLE_OPS = ['pageload', 'navigation', 'ui.load'];
+
+export type RepresentativeTraceEvent = {
+  event: TraceTree.TraceEvent | OurLogsResponseItem | null;
+  type: 'trace' | 'log';
+};
+
+export const getRepresentativeTraceEvent = (
+  tree: TraceTree,
+  logs: OurLogsResponseItem[] | undefined
+): RepresentativeTraceEvent => {
+  const hasLogs = logs && logs.length > 0;
+  if (tree.type === 'empty' && hasLogs) {
+    return {
+      event: logs[0]!,
+      type: 'log',
+    };
+  }
+
+  const traceNode = tree.root.children[0];
+
+  if (!traceNode) {
+    return {
+      event: null,
+      type: 'trace',
+    };
+  }
+
+  if (!isTraceNode(traceNode) && !isEAPTraceNode(traceNode)) {
+    throw new TypeError('Not trace node');
+  }
+
+  let firstRootEvent: TraceTree.TraceEvent | null = null;
+  let candidateEvent: TraceTree.TraceEvent | null = null;
+  let firstEvent: TraceTree.TraceEvent | null = null;
+
+  const events = isTraceNode(traceNode)
+    ? [...traceNode.value.transactions, ...traceNode.value.orphan_errors]
+    : traceNode.value;
+  for (const event of events) {
```

Comment:
nit: this for loops feels a little hard to read. You might have an easier time using something like to the following:
```
const firstRootEvent = events.find(isRootEvent);
```
And something similar for `candidateEvent` and `firstEvent`.